dubbo-rpc/dubbo-rpc-dubbo/src/main/java/org/apache/dubbo/rpc/protocol/dubbo/telnet/InvokeTelnetHandler.java

```diff
@@ -75,6 +76,14 @@ private static boolean isMatch(Class<?>[] types, List<Object> args) {
             }
 
             if (ReflectUtils.isPrimitive(arg.getClass())) {
+                if (arg instanceof String && type.isEnum()) {
```

Comment:
I think it is not necessary to call `CompatibleTypeUtils.compatibleTypeConvert(arg, type`, this logic can be simplified as below, since we know arg is always a String type in this case.

```java
            if (ReflectUtils.isPrimitive(arg.getClass())) {
                if (!ReflectUtils.isPrimitive(type) && !type.isEnum()) {
                    return false;
                }
            }
```